vault/rekey.go

```diff
@@ -674,23 +674,23 @@ func (c *Core) RecoveryRekeyUpdate(ctx context.Context, key []byte, nonce string
 		return nil, logical.CodedError(http.StatusBadRequest, fmt.Errorf("recovery key verification failed: %w", err).Error())
 	}
 
-	// Generate a new master key
-	newMasterKey, err := c.barrier.GenerateKey(c.secureRandomReader)
+	// Generate a new root key
```

Comment:
I think many of the master/root key names in this little section should probably be unseal instead, since I see that we're splitting with shamir.

Reply:
My mistake, not unseal, recovery.